test/WatchTestCases.test.js

```diff
@@ -0,0 +1,175 @@
+			category.tests.forEach(function(testName) {
+				describe(testName, function() {
+					var tempDirectory = path.join(__dirname, "js", "watch-src", category.name, testName);
+					var testDirectory = path.join(casesPath, category.name, testName);
+					var runs = fs.readdirSync(testDirectory).sort().map(function(name) {
+						return {
+							name: name,
+							suite: describe(name, function() {})
+						}
+					});
+					before(function() {
+						remove(tempDirectory);
+					});
+					it("should compile", function(done) {
+						this.timeout(30000);
+						var outputDirectory = path.join(__dirname, "js", "watch", category.name, testName);
+
+						var options = {
+							context: tempDirectory,
+							entry: "./index.js",
+							output: {
+								path: outputDirectory,
+								pathinfo: true,
+								filename: "bundle.js"
+							}
+						};
+
+						var runIdx = 0;
+						var run = runs[runIdx];
+						var lastHash = "";
+						copyDiff(path.join(testDirectory, run.name), tempDirectory);
+
+						var compiler = webpack(options);
+						compiler.watch({}, function(err, stats) {
+							if(stats.hash === lastHash)
+								return;
+							lastHash = stats.hash;
+							if(run.done)
+								return done(new Error("Compilation changed but no change was issued " + lastHash + " != " + stats.hash + " (run " + runIdx + ")"));
+							run.done = true;
+							if(err) return done(err);
+							var statOptions = Stats.presetToOptions("verbose");
+							statOptions.colors = false;
+							fs.writeFileSync(path.join(outputDirectory, "stats.txt"), stats.toString(statOptions), "utf-8");
+							var jsonStats = stats.toJson({
+								errorDetails: true
+							});
+							if(checkArrayExpectation(testDirectory, jsonStats, "error", "Error", done)) return;
+							if(checkArrayExpectation(testDirectory, jsonStats, "warning", "Warning", done)) return;
+							var exportedTests = 0;
+
+							function _it(title, fn) {
+								var test = new Test(title, fn);
+								run.suite.addTest(test);
+								exportedTests++;
+								return test;
+							}
+
+							function _require(currentDirectory, module) {
+								if(Array.isArray(module) || /^\.\.?\//.test(module)) {
+									var fn;
+									var content;
+									if(Array.isArray(module)) {
+										var p = path.join(currentDirectory, module[0]);
+										content = module.map(function(p) {
+											var p = path.join(currentDirectory, p);
+											return fs.readFileSync(p, "utf-8");
+										}).join("\n");
+									} else {
+										var p = path.join(currentDirectory, module);
+										content = fs.readFileSync(p, "utf-8");
+									}
+									fn = vm.runInThisContext("(function(require, module, exports, __dirname, __filename, it, WATCH_STEP) {" + content + "\n})", p);
+									var module = {
+										exports: {}
+									};
+									fn.call(module.exports, _require.bind(null, path.dirname(p)), module, module.exports, path.dirname(p), p, _it, run.name);
```

Comment:
Maybe its line 141 thats the one in question. But do you have to pass the required `should` in as a variable in the `vm.runInThisContext()`, would make sense why `it` works but not `should`. This is just a hunch, tbh.